docs/federate.md

```diff
@@ -0,0 +1,156 @@
+
+For both methods let's say you want to run your server at
+``synapse.example.com`` on port ``443`` (instead of ``8448``), but you
+want to have your Matrix user-ids look like ``@user:example.com``.
+
+Without configuring delegation, the matrix federation will
+expect to find your resources via ``example.com:8448``. The
+following methods allow you to provide a different server and port for
+``*:example.com`` resources.
+
+If all goes well, you should be able to [connect to your server with a client](../README.rst#registering-a-new-user-from-a-client).
+and then join a room via federation. (A good place to start is ``#synapse:matrix.org`` - a room for 
+Synapse admins.)
+
+
+DNS SRV delegation
+------------------
+
+To use this method, you need to have write access to your
+``server_name`` 's domain zone DNS records (in our example it would be
+``example.com`` DNS zone).
+
+This method additionally requires your delegate server to provide a
+valid SSL certificate identifying it as the original ``server_name``
+domain zone. So the delegate domain name is only used to resolve a possible 
+different IP/Port combination to find your server. You must use the other 
+delegation method is this isn't what you want. (for more details on the 
+rationale [see here](https://github.com/matrix-org/matrix-doc/blob/master/proposals/1711-x509-for-federation.md#interaction-with-srv-records>))
+
+You need to add a SRV record in your ``server_name`` 's DNS zone with
+this format::
+
+     _matrix._tcp.<yourdomain.com> <ttl> IN SRV 10 0 <port> <synapse.server.name>
+
+In our example, we would need to add this SRV record in the
+``example.com`` DNS zone::
+
+     _matrix._tcp.example.com. 3600 IN SRV 10 0 443 synapse.example.com.
+
+
+Once done and set up, you can check the DNS record with ``dig -t srv
+_matrix._tcp.<server_name>``, in our example, we would expect this::
+
+    $ dig -t srv _matrix._tcp.example.com
+    _matrix._tcp.example.com. 3600    IN      SRV     10 0 443 synapse.example.com.
+
+Note that the server host name cannot be an alias (CNAME record): it has to point
+directly to the server hosting the synapse instance.
+
+
+.well-known delegation
+----------------------
+
+To use this method, you need to be able to alter the
+``server_name`` 's https server to serve the ``/.well-known/matrix/server`` 
+URL. Having an active server (with a valid ``SSL`` certificate) serving your 
+``server_name`` domain is out of the scope of this documentation.
+
+The URL ``https://<server_name>/.well-known/matrix/server`` should
+return a JSON structure containing the key ``m.server`` like so::
+
+    {
+	    "m.server": "<synapse.server.name>:<yourport>"
+    }
+
+In our example, this would mean that URL ``https://example.com/.well-known/matrix/server``
+should return::
+
+    {
+	    "m.server": "synapse.example.com:443"
+    }
+
+This delegation method allows a full delegation contrary to the DNS SRV
```

Comment:
I think this paragraph needs a rethink, but I'm not quite sure what it should look like.

Reply:
What are your concerns?

Reply:
Can you explain your concerns?

Reply:
Addressed our of band and changes committed - wanted to ensure that the methods were considered 'different' rather than one being preferable to another.